libs/ui/src/hooks/ui_strings_api.ts

```diff
@@ -0,0 +1,46 @@
+import { QueryClient, QueryKey, useQuery } from '@tanstack/react-query';
+import { UiStringsApi } from '@votingworks/types';
+import * as grout from '@votingworks/grout';
+
+// Unable to use `grout.Client<UseStringsApi>` directly due to some mismatched
+// type inference with `grout.AnyMethods`, so copying the `grout.Client`
+// definition here for now:
+export type UiStringsApiClient = {
+  [Method in keyof UiStringsApi]: grout.AsyncRpcMethod<UiStringsApi[Method]>;
+};
+
+function createReactQueryApi(getApiClient: () => UiStringsApiClient) {
+  return {
+    getAvailableLanguages: {
+      getQueryKey(): QueryKey {
+        return ['getAvailableLanguages'];
+      },
+
+      useQuery() {
+        const apiClient = getApiClient();
+
+        return useQuery(this.getQueryKey(), () =>
+          apiClient.getAvailableLanguages()
+        );
+      },
+    },
+
+    async onMachineConfigurationChange(
+      queryClient: QueryClient
```

Comment:
For consistency, would it make sense to pass the `queryClient` (or `getQueryClient`) as a top-level input to `createReactQueryApi` / `createUiStringsApi`?

```ts
createReactQueryApi(
  getApiClient: () => UiStringsApiClient,
  getQueryClient: () => QueryClient
)
```

So that all methods (`getAvailableLanguages`, `onMachineConfigurationChange`) always have access to both without them needing to be passed in.

I'm okay with the current approach, too - maybe there's a reason to differentiate the `apiClient` from the `queryClient` that I'm missing

Reply:
Ah yeah, +1 to consistency!

Reply:
Bah, looks like we're not allowed to invoke `useQueryClient` inside of the `onSuccess` handler of mutations (since it's outside of the component context, so that forces us to pass in the query client to the change handler at execution time. Will drop the latest commit for now

Reply:
Ohh I see, makes sense - leaving as is is fine then!